pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java

```diff
@@ -328,7 +319,20 @@ void refreshSegmentStates() {
       }
     }
 
-    _segmentStates = new SegmentStates(instanceCandidatesMap, unavailableSegments);
+    _segmentStates = new SegmentStates(instanceCandidatesMap, servingInstances, unavailableSegments);
+  }
+
+  private List<SegmentInstanceCandidate> getEnabledCandidates(List<SegmentInstanceCandidate> candidates,
```

Comment:
Add some comments since this method is doing more than the method name or fix the method name?

Reply:
Done